homeassistant/components/media_player/emby.py

```diff
@@ -96,6 +97,37 @@ def device_update_callback(data):
             _LOGGER.debug("Adding new devices: %s", new_devices)
             async_add_devices(new_devices, update_before_add=True)
 
+        if auto_group:
+            # Compile a list of entity IDs for active emby instances
+            active_entity_ids = []
+
+            for emby_device_id, device in active_emby_devices.items():
+
+                _LOGGER.debug("Indexing active emby device %s", device.name)
+                _LOGGER.debug("Indexing active emby device ID %s", emby_device_id)
+
+                # Reformat device name into the entity ID format used by
+                # HomeAssistant if we don't do this, we won't be able to
+                # create a group of these entities
+                entity_name = device.name.replace(" - ", "__")
+                entity_name = entity_name.replace(" ", "_")
+                # Prepend media_player prefix used by HomeAssistant -
+                # this is how it would appear in the UI
+                entity_name = "media_player." + entity_name
+                # Convert to lower-case for convention purposes
+                entity_name = entity_name.lower()
+                # Add to list
+                active_entity_ids.append(entity_name)
+
+            # Remove the default device from the group
+            active_entity_ids.remove(
+                "media_player.emby__homeassistant__homeassistant")
```

Comment:
The pyemby library should be automatically excluding the homeassistant device from being displayed making this not necessary.  Are you seeing different behavior?

Reply:
Hi @mezz64,

Having played around with it some more... I am, but not consistently. Sometimes the homeassistant device appears, sometimes it doesn't. I'm going to change the code to check if the entity is there first, before removing it.

I haven't looked into it too much at this stage so I couldn't tell you much more than sometimes I see it, sometimes I don't.

Reply:
Can you check and see if you have two homeassistant devices displaying in your Emby dashboard on the instances when you see it show up in homeassistant?

pyEmby generates a unique ID for the server connection on each start, it then excludes the device  with that ID from being discovered by the library.  My guess is the times you see it show up in homeassistant an orphaned device is hanging around from an old session.

Reply:
I don't believe there were two, but I will have a play around with it today and let you know what I see.